File: quizzes/quizzesV2/quizzes10/11-problem.ts
export {};

const isConstructed = (str1: string, str2: string): boolean => {
    const counts = new Map<string, number>();

    for (const char of str1) {
        counts.set(char, (counts.get(char) ?? 0) + 1);
    }

    for (const char of str2) {
        const currentCount = counts.get(char) ?? 0;
        if (currentCount <= 0) return false;
        counts.set(char, currentCount - 1);
    }

    return true;
};

console.log(isConstructed('wwcdore', 'coder')); // "true"
console.log(isConstructed('h3llko', 'hello')); // "false"
console.log(isConstructed('rkqodlw', 'world')); // "true"

File: bins/binsV2/bin/bins2/11-problem.rb
# frozen_string_literal: true

require 'json'

def is_constructed(str1, str2)
  counts = Hash.new(0)

  str1.each_char { |c| counts[c] += 1 }

  str2.each_char do |c|
    return false if counts[c] <= 0
    counts[c] -= 1
  end

  true
end

puts is_constructed('wwcdore', 'coder')  # true
puts is_constructed('h3llko', 'hello')   # false
puts is_constructed('rkqodlw', 'world')  # true
